src/Config/SessionConfig.php

```diff
@@ -447,40 +447,39 @@ private function locateRegisteredSaveHandlers()
      */
     private function performSaveHandlerUpdate($phpSaveHandler)
     {
-        $knownHandlers = $this->locateRegisteredSaveHandlers();
-
-        if (in_array($phpSaveHandler, $knownHandlers, true)) {
-            $phpSaveHandler = strtolower($phpSaveHandler);
-            set_error_handler([$this, 'handleError']);
-            session_module_name($phpSaveHandler);
-            restore_error_handler();
-            if ($this->phpErrorCode >= E_WARNING) {
+        if (is_string($phpSaveHandler)) {
+            $knownHandlers = $this->locateRegisteredSaveHandlers();
+
+            if (in_array($phpSaveHandler, $knownHandlers, true)) {
+                $phpSaveHandler = strtolower($phpSaveHandler);
+                set_error_handler([$this, 'handleError']);
+                session_module_name($phpSaveHandler);
+                restore_error_handler();
+                if ($this->phpErrorCode >= E_WARNING) {
+                    throw new Exception\InvalidArgumentException(sprintf(
+                        'Error setting session save handler module "%s": %s',
+                        $phpSaveHandler,
+                        $this->phpErrorMessage
+                    ));
+                }
+
+                return $phpSaveHandler;
+            }
+
+            if ((! class_exists($phpSaveHandler)
```

Comment:
we can remove one parentheses, these are redundant

Reply:
this is fixed.